### ISSUE TITLE
-check option on copy command (enhancement)

### ISSUE THREAD

**Issue by TowerBR** [gilbertchen/duplicacy#385]
This is an **enhancement**, based on [this topic](https://duplicacy.com/issue?id=5673497447628800) on Duplicacy forum.

It's about adding a ```-check``` option to the ```copy``` command, so it would check the chunks, not just the snapshots.

**Comment by TheBestPessimist** [comment]
I still don't understand what `-check` should actually do. Could you please explain a little bit, @TowerBR ?

**Comment by gilbertchen** [comment]
The current behavior is that when a snapshot file already exists in the destination storage, Duplicacy will assume that all chunks referenced by this snapshot should exist in the destination storage too.  The proposed `-check` option would force Duplicacy to not make this assumption and copy such chunks anyway in this case.

**Comment by TheBestPessimist** [comment]
So this basically fixes the problem if a chunk was lost by accident or some other problem?

**Comment by gilbertchen** [comment]
Right.  It is equivalent to running a `check` command after the copy.

**Comment by TheBestPessimist** [comment]
But it is not equivalent: i was expecting that `backup -check` ignores the "snapshot-already-existing-files" and actually tries to upload everything (each chunk) again, so that we are sure that the current latest backup REALLY-100%HAS-ALL-THE-CHUNKS.

Basically, i understand that this would be the simple solution for the [Missing Chunks](../wiki/Missing-Chunks) problem.

**Comment by TheBestPessimist** [comment]
Here are the issues that i found, which are related to this (that is: if my understanding of this flag is correct): #234 #306 #335 and #385 (this one).

**Comment by TowerBR** [comment]
> The proposed -check option would force Duplicacy to not make this assumption and copy 
> such chunks anyway in this case.

Exactly!

> May i suggest the flag is named -ignore-shapshot-cache?

```-ignore-shapshot-cache``` seems to be referring to the snapshots, and the problem here is to "deepen" the scan until the chunks.

Maybe it can be ```-exhaustive```, as in ```prune``` command.

**By the way: I just noticed that the linke in the first post was wrong, I just corrected it.**

**Comment by TheBestPessimist** [comment]
Ah, now that the link is corrected, i see that the issues are totally different: i was thinking of the `backup` command while you were actually talking about `copy`. 

In any case, should i comment in

**Comment by TowerBR** [comment]
I liked the ```-check-all-chunks``` name you proposed in [#306](https://github.com/gilbertchen/duplicacy/issues/306), is simple and straightforward, with no interpretation problems.

**Comment by TheBestPessimist** [comment]
Since naming is hard (for this one programmer at least) `-check-all-chunks` implies (again) the same problem as with `-exhaustive` mentioned above. On the other hand it is much better than `-exhaustive` :).

Hence, this is the reason i came up with the `*-snapshot-chunks` idea (`-check-snapshot-chunks`).

Anyhow, i'll let you 2 chose the name coz i think i'm becoming a bit of a pest -- and that is never cool.

**Comment by jt70471** [comment]
To clarify, you're requesting that all chunks in the source location are verified to exist in the destination location, not that the chunk contents themselves match, correct? Doing an actual comparison of the chunk contents in both source and destination would be extremely costly for elapsed run time and data transferred.

**Comment by TowerBR** [comment]
Exactly, check if the chunks are there, and not just check the snapshots.
